### PR TITLE
chore(monitoring): add opencensus samples

### LINKED ISSUE #5342: Add example for using OpenCensus with Prometheus exporter
## In which file did you encounter the issue?

None yet

### Did you change the file? If so, how?

No.

## Describe the issue

This is a tracking issue for the work I'm doing to add specific examples to the /monitoring folder.  This example will show how to instrument a Flask application with OpenCensus metrics (using the Prometheus exporter) to write SLIs.  It's specifically in support of the upcoming SLI documentation (internal to Google cl/343542215).

### PR DESCRIPTION
**Pull request by yuriatgoogle**
## Description

Fixes #5342

Note: It's a good idea to open an issue first for discussion.

## Checklist
- [ V] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md)
- [ V] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md#readme-file)
- [ V] **Tests** pass:   `nox -s py-3.6` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md#test-environment-setup))
- [ V] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md#test-environment-setup))
- [ x] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ x] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ V] Please **merge** this PR for me once it is approved.
- [ x] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/.github/CODEOWNERS) with the codeowners for this sample

### CODE REVIEW

File: monitoring/opencensus/.python-version
Comment (busunkim96) [inline monitoring/opencensus/.python-version:1]: Is this file consumed by open census?

File: monitoring/opencensus/noxfile.py
Comment (busunkim96) [inline monitoring/opencensus/noxfile.py:1]: Please delete this file. CI will copy it into the directory from the base of the repository.

File: monitoring/opencensus/noxfile_config.py
Comment (busunkim96) [inline monitoring/opencensus/noxfile_config.py:28]: I don't see type hints in the sample files. Did you intend for this to be set to `True`?

File: monitoring/opencensus/requirements.txt
Comment (busunkim96) [inline monitoring/opencensus/requirements.txt:1]: Are all of these requirements needed by the sample? Please delete any that the sample does not explicitly need to function.

File: monitoring/opencensus/.python-version
Comment (yuriatgoogle) [reply monitoring/opencensus/.python-version:1]: No, it's created by `pyenv`. Going to delete it.

File: monitoring/opencensus/noxfile.py
Comment (yuriatgoogle) [reply monitoring/opencensus/noxfile.py:1]: Done, thank you.

File: monitoring/opencensus/noxfile_config.py
Comment (yuriatgoogle) [reply monitoring/opencensus/noxfile_config.py:28]: No, that was not intended, fixing.

File: monitoring/opencensus/requirements.txt
Comment (yuriatgoogle) [reply monitoring/opencensus/requirements.txt:1]: Pruned, thank you.

**Comment by yuriatgoogle** [comment]
I am so sorry... this is a mess right now.  I think I accidentally merged `master` into this branch, and now I am struggling to get back to a good state where I can actually make the requested changes.

**Comment by yuriatgoogle** [comment]
Meeting with Ace shortly to resolve all this merge nonsense.

File: monitoring/opencensus/requirements.txt
Comment (busunkim96) [inline monitoring/opencensus/requirements.txt:13]: I think there are a few more that could be deleted:
* `flake8` - only needed for linting
* `nox` - the test runner
* `pytest` - for testing

File: monitoring/opencensus/requirements-test.txt
Comment (busunkim96) [inline monitoring/opencensus/requirements-test.txt:1]: I see `flask` here and `Flask` in the other file. Are they different packages? 

`requirements-test.txt` should have the additional requirements for the tests (that aren't needed to run the sample).  `pytest` is usually enough.

File: monitoring/opencensus/noxfile_config.py
Comment (busunkim96) [inline monitoring/opencensus/noxfile_config.py:1]: Apologies, I should have mentioned this in my earlier review. This file only needs to be added if you want to change the default settings. It looks like the defaults are fine, so please delete this file.

Out of curiosity, were you going off of an existing sample when you created this directory or following the Authoring Guide? We're revamping our documentation so I'd love to know which docs that we're surfacing (or not surfacing). :)

File: monitoring/opencensus/README.rst.in
Comment (busunkim96) [inline monitoring/opencensus/README.rst.in:1]: Thanks for adding this! Please also trigger the README generation:

1. Install nox. `pip install nox`.
2. Temporarily copy the noxfile from the root of the repo into this directory. `cp ../../noxfile-template.py noxfile.py`
3. `nox -s readmegen` (this should create a README.rst)
4. Delete the `noxfile.py`.

File: monitoring/opencensus/main.py
Comment (busunkim96) [inline monitoring/opencensus/main.py:62]:
```suggestion
failed_request_count_view = view_module.View(
```

File: monitoring/opencensus/main.py
Comment (busunkim96) [inline monitoring/opencensus/main.py:72]: Please use `camel_case` for the function names. https://google.github.io/styleguide/pyguide.html#316-naming

**Review by busunkim96 — CHANGES_REQUESTED** [review]
If this is intended for cloud.google.com, you may want to include region tags. https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md#region-tags

**Comment by ace-n** [comment]
BTW @busunkim96 any need to add this to `CODEOWNERS`?

File: monitoring/opencensus/requirements.txt
Comment (yuriatgoogle) [reply monitoring/opencensus/requirements.txt:13]: Done, thank you.

File: monitoring/opencensus/requirements-test.txt
Comment (yuriatgoogle) [reply monitoring/opencensus/requirements-test.txt:1]: I've removed this from here, I think it was a mistake - thank you.

File: monitoring/opencensus/requirements.txt
Comment (yuriatgoogle) [reply monitoring/opencensus/requirements.txt:13]: RE: region tags - I may need to add those once the Cloud Docs folks start to incorporate this into the doc, but thank you for pointing that out.

File: monitoring/opencensus/noxfile_config.py
Comment (yuriatgoogle) [reply monitoring/opencensus/noxfile_config.py:1]: Deleted noxfile_config.py.

I was... kind of trying to do both? I used the Authoring Guide heavily and also ended up copying some stuff from existing samples, like things in the Dataflow directory.  This is my first contribution, and I'm not exactly a Python expert, so I likely made some obvious mistakes.

I think that a "how to send your first PR" or something like that would be helpful - almost like a codelab?  I think the Authoring Guide does a mostly good job, but it has a lot of guidelines that make the entire doc a bit unwieldy.  Someone like me could really use a dedicated guide on the mechanics, separate from the style guidelines.

File: monitoring/opencensus/README.rst.in
Comment (yuriatgoogle) [reply monitoring/opencensus/README.rst.in:1]: I'm trying to do that now - running `nox -s readmegen` results in this:

```
$ nox -s readmegen
No user noxfile_config found: detail: No module named 'noxfile_config'
nox > Running session readmegen(path=PosixPath('README.rst.in'))
nox > Creating virtual environment (virtualenv) using python3.9 in .nox/readmegen-path-posixpath-readme-rst-in
nox > python -m pip install jinja2 pyyaml
nox > python -m pip install -r requirements.txt
nox > python /Users/ygrinshteyn/src/forks/python-docs-samples/scripts/readme-gen/readme_gen.py README.rst.in
/Users/ygrinshteyn/src/forks/python-docs-samples/scripts/readme-gen/readme_gen.py:54: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  config = yaml.load(f)
Dropping 1 non-positive bucket boundaries
 * Running on http://127.0.0.1:8080/ (Press CTRL+C to quit)
```

It just seems to hang here forever...

File: monitoring/opencensus/main.py
Comment (yuriatgoogle) [reply monitoring/opencensus/main.py:62]: Thank you for catching this! Fixed.

File: monitoring/opencensus/main.py
Comment (yuriatgoogle) [reply monitoring/opencensus/main.py:72]: Fixed, thank you!

File: monitoring/opencensus/README.rst.in
Comment (yuriatgoogle) [reply monitoring/opencensus/README.rst.in:1]: This is what happens when I interrupt it:

```
Traceback (most recent call last):
  File "/Users/ygrinshteyn/src/forks/python-docs-samples/scripts/readme-gen/readme_gen.py", line 66, in <module>
    main()
  File "/Users/ygrinshteyn/src/forks/python-docs-samples/scripts/readme-gen/readme_gen.py", line 59, in main
    output = README_TMPL.render(config)
  File "/Users/ygrinshteyn/src/forks/python-docs-samples/monitoring/opencensus/.nox/readmegen-path-posixpath-readme-rst-in/lib/python3.9/site-packages/jinja2/environment.py", line 1088, in render
    return concat(self.root_render_func(self.new_context(vars)))
  File "/Users/ygrinshteyn/src/forks/python-docs-samples/scripts/readme-gen/templates/README.tmpl.rst", line 98, in root
  File "/Users/ygrinshteyn/src/forks/python-docs-samples/monitoring/opencensus/.nox/readmegen-path-posixpath-readme-rst-in/lib/python3.9/site-packages/jinja2/runtime.py", line 290, in call
    return __obj(*args, **kwargs)
  File "/Users/ygrinshteyn/src/forks/python-docs-samples/scripts/readme-gen/readme_gen.py", line 37, in get_help
    return subprocess.check_output(['python', file, '--help']).decode()
  File "/Users/ygrinshteyn/.pyenv/versions/3.9.1/lib/python3.9/subprocess.py", line 420, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/Users/ygrinshteyn/.pyenv/versions/3.9.1/lib/python3.9/subprocess.py", line 503, in run
    stdout, stderr = process.communicate(input, timeout=timeout)
  File "/Users/ygrinshteyn/.pyenv/versions/3.9.1/lib/python3.9/subprocess.py", line 1117, in communicate
    stdout = self.stdout.read()
KeyboardInterrupt
nox > Interrupted...
nox > Session readmegen(path=PosixPath('README.rst.in')) interrupted.
```

**Comment by busunkim96** [comment]
@ace-n Yep that's a good point, please add a new line to https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/.github/CODEOWNERS with the owners for this directory.

File: monitoring/opencensus/README.rst.in
Comment (busunkim96) [reply monitoring/opencensus/README.rst.in:1]: Huh strange. 🤔  We're kind of moving away from the generated README.rsts in their current form. Would you mind re-working this to be a standalone README.md? See https://github.com/GoogleCloudPlatform/python-docs-samples/tree/master/appengine/standard for an example.

Comment (yuriatgoogle) [reply monitoring/opencensus/README.rst.in]: Done.

**Comment by yuriatgoogle** [comment]
> @ace-n Yep that's a good point, please add a new line to https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/.github/CODEOWNERS with the owners for this directory.

Done.

**Comment by yuriatgoogle** [comment]
@busunkim96 

> If this is intended for cloud.google.com, you may want to include region tags. https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md#region-tags

I've added region tags - please have another look and approve and merge if everything is OK?

**Review by ace-n — APPROVED** [review]
LGTM, @busunkim96 please approve. 🙂

**Review by busunkim96 — APPROVED** [review]

**Pull request merged**
